File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m = 0;
        int minValue = prices[0];
        for(int i=1;i<prices.size();i++){
            m = max(m, prices[i] - minValue);
            minValue = min(minValue, prices[i]);
        }
        return m;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;
        int best = 0;
        for(int i=0;i<prices.length;i++){
            max = Math.max(max, prices[i]);
            best = Math.max(best, max - min);
            if(min > prices[i]){
                min = prices[i];
                max = prices[i];
            }
        }
        return best;
    }
}